python/matrix/matrix.py

```diff
@@ -5,15 +5,10 @@ def __init__(self, matrix_string):
         for i in lines:
```

Comment:
Storing everything in separate variables may be useful while you try to figure out what to do.
In simple cases like this one, try to inline the lines object afterwards, i.e. replace its only usage with its definition.